apps/x509.c

```diff
@@ -477,18 +489,32 @@ int x509_main(int argc, char **argv)
         goto end;
     }
 
+    if (newcert && fkeyfile == NULL) {
+        BIO_printf(bio_err, "We need a public key, use -force_pubkey or -in\n");
+        goto end;
+    }
     if (fkeyfile != NULL) {
         fkey = load_pubkey(fkeyfile, keyformat, 0, NULL, e, "Forced key");
         if (fkey == NULL)
             goto end;
     }
 
+    if (newcert && subj == NULL) {
+        BIO_printf(bio_err, "We need a subject, use -subj or -in\n");
```

Comment:
The error says `use -subj or -in`, but the test seems to require only `-subj`. Is this right?

Reply:
Hmm, this was really a bit fishy, and the error on `-new` without `-force_pubkey`, too.
The idea was that `-in` (at least with `-req`) could also provide a subject/pubkey, but `-new` together with `-in` would be rather unclean. 

So I've removed the "or -in" in both error messages and made sure that `-in` is ignored (with a warning) when `-new` is present.